crates/goose/src/providers/openai.rs

```diff
@@ -134,23 +135,84 @@ impl Provider for OpenAiProvider {
         messages: &[Message],
         tools: &[Tool],
     ) -> Result<(Message, ProviderUsage), ProviderError> {
-        let payload = create_request(&self.model, system, messages, tools, &ImageFormat::OpenAi)?;
-
-        // Make request
-        let response = self.post(payload.clone()).await?;
-
-        // Parse response
-        let message = response_to_message(response.clone())?;
-        let usage = match get_usage(&response) {
-            Ok(usage) => usage,
-            Err(ProviderError::UsageError(e)) => {
-                tracing::debug!("Failed to get usage data: {}", e);
-                Usage::default()
+        let use_tool_shim = std::env::var("GOOSE_TOOLSHIM")
+            .map(|val| val == "1" || val.to_lowercase() == "true")
+            .unwrap_or(false);
+
+        if use_tool_shim {
```

Comment:
i wonder what's the best place to include this logic. thinking out loud (i'm not convinced if this is a good approach):

- we can have some options in the config (say `structure_chat_tool_calls`, model config for structuring)
- based on those, in the agent loop we decide how to construct the msgs and the tools when calling [provider.complete()](https://github.com/block/goose/blob/4283b2171325b0800649385e91072415580f4213/crates/goose/src/agents/truncate.rs#L175-L179)
- we'll need another method on the provider to `structure_responses` which also gets called in the agent based on the config options